pwiz_tools/Skyline/Model/ImportPeakBoundaries.cs

```diff
@@ -100,26 +101,57 @@ public enum Field { modified_peptide, filename, apex_time, start_time, end_time,
 
         // ReSharper disable LocalizableElement
 
+        private static string[] _peptide_synonyms; // Cache for performance reasons (L10N stuff is expensive)
         public static string[] PEPTIDE_SYNONYMS
         {
             get
             {
-                return new[] 
+                if (_peptide_synonyms == null)
                 {
-                    "PeptideModifiedSequence", "ModifiedSequence", "FullPeptideName", "EG.ModifiedSequence",
-                    ColumnCaptions.PeptideModifiedSequence, ColumnCaptions.ModifiedSequence
-                };
+                    var headers = new List<string>()
+                    {
+                        "PeptideModifiedSequence", "ModifiedSequence", "FullPeptideName", "EG.ModifiedSequence",
+                    };
+                    // Also recognize column captions in all supported display languages
+                    var currentUICulture = Thread.CurrentThread.CurrentUICulture;
+                    foreach (var culture in CultureUtil.AvailableDisplayLanguages())
+                    {
+                        Thread.CurrentThread.CurrentUICulture = culture;
+                        headers.Add(ColumnCaptions.PeptideModifiedSequence);
+                        headers.Add(ColumnCaptions.ModifiedSequence);
+                    }
+
+                    Thread.CurrentThread.CurrentUICulture = currentUICulture;
+                    _peptide_synonyms = headers.ToArray();
+                }
+                return _peptide_synonyms;
             }
         }
 
+        private static string[] _molecule_synonyms; // Cache for performance reasons (L10N stuff is expensive)
         public static string[] MOLECULE_SYNONYMS
         {
             get
             {
-                return new[] 
+                if (_molecule_synonyms == null) 
                 {
-                    "Molecule", "MoleculeName", ColumnCaptions.Molecule, ColumnCaptions.MoleculeName
-                };
+                    var headers = new List<string>()
+                    {
+                        "Molecule", "MoleculeName"
+                    };
+                    // Also recognize column captions in all supported display languages
+                    var currentUICulture = Thread.CurrentThread.CurrentUICulture;
+                    foreach (var culture in CultureUtil.AvailableDisplayLanguages())
+                    {
+                        Thread.CurrentThread.CurrentUICulture = culture;

```

Comment:
You can use CurrentCultureSetter here instead.

Reply:
Hmm, that doesn't set the UI culture. Is that a bug, or should I make a CurrentCultureUISetter class as well?

Reply:
I always use the method "LocalizationHelper.CallWithCulture" instead of "CurrentCultureSetter".